packages/algob/src/builtin-tasks/deploy.ts

```diff
@@ -94,5 +101,5 @@ export default function (): void {
       "A directory that contains js files to be run within algob's environment",
       []
     )
-    .setAction(doDeploy);
+    .setAction(async (input, env) => await executeDeployTask(input, env, createDeployClient));
```

Comment:
no need to use async / await here.

Reply:
Also, instead of this, maybe it's easier and more readable to have 3 functions:
```
executeRun(input, env) => return executeTask(input, env, createRunClient())
executeDeploy(input, env) => return executeTask(input, env, createDeployClient())
executeTask(input, env, deployer) ...
```

Reply:
The tasks are very different. It's not client that differs, it's the logic of `executeDeployTask` that differs very much. I already tried to reuse as much as I could.

Removed async + await from both tasks.

Reply:
This is not a valid typescript syntax:
```
executeDeploy(input, env) => return executeTask(input, env, createDeployClient())
```
It's either `(input, env) => return executeTask(input, env, createDeployClient())` which doesn't make sense or 
`(input, env) => return executeDeployTask(input, env, createDeployClient())` which is actually the original code.

This doesn't make sense to me too:
```
executeDeploy(
  (input, env) => return executeTask(input, env, createDeployClient())
)
```

This may make sense because it's what we've had before but that's simply code duplication for the sake of total purity:
```
const executeDeploy1 = (input, env) => return executeDeploy(input, env, createDeployClient())
...
.setAction((input, env) => return executeDeploy1(input, env))

----

const executeRun1 = (input, env) => return executeRun(input, env, createDeployClient())
...
.setAction((input, env) => return executeRun1(input, env))
```

Reply:
> This is not a valid typescript syntax:

I know - it's a pseudocode, but you see the idea.

Reply:
I prefer 
```
(input, env) => executeDeployTask(input, env, createDeployClient())  // return not needed if {...} is ommited
```

Reply:
Well this is what I have in master:
![Screenshot at 2020-08-31 14-31-26](https://user-images.githubusercontent.com/1641263/91715697-d9356b80-eb96-11ea-9a36-5b546c810cc3.png)
It was the initial code and async+await was removed long ago.